.github/workflows/release.yml

```diff
@@ -374,3 +370,63 @@ jobs:
         run: |
           python -m pip install twine
           venv\Scripts\python misc/upload_release.py
+
+  create_release:
+    name: Create tag and publish release
+    needs:
+      [
+        build_and_upload_linux,
+        build_and_upload_mac,
+        build_and_upload_m1,
+        build_and_upload_macos_1014,
+        build_and_upload_windows,
+      ]
+    runs-on: ubuntu-latest
+    if: github.event_name == 'workflow_dispatch'
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Generate Changelog
+        id: changelog
+        run: |
+          pip3 install gitpython
+          content=$(python3 misc/make_changelog.py)
+          echo $content
+          # Escape multiline strings:
+          # https://renehernandez.io/snippets/multiline-strings-as-a-job-output-in-github-actions/
+          content="${content//'%'/'%25'}"
+          content="${content//$'\n'/'%0A'}"
+          content="${content//$'\r'/'%0D'}"
+          echo "::set-output name=content::$content"
+
+      - name: Create tag
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git tag -a ${RELEASE_VERSION} -m "Release ${RELEASE_VERSION}"
+          git push origin --tags
+
+      - name: Publish release
+        uses: softprops/action-gh-release@v1
+        with:
+          body: ${{ steps.changelog.outputs.content }}
+          tag_name: ${{ github.event.inputs.version }}
+
+      - name: Bump version
+        run: |
+          import os
+          version = os.getenv("RELEASE_VERSION")
+          prefix, patch = version.rsplit(".", 1)
+          next_version = f"{prefix}.{int(patch) + 1}"
+          with open('version.txt', 'w') as f:
+              f.write(next_version + '\n')
+        shell: python
+
+      - name: Commit and Push Changes
+        run: |
+          git add version.txt
+          git commit -m "[misc] Bump version to $(cat version.txt)"
+          git push origin master
```

Comment:
Since we have turned on the branch protection, would this still work?

Reply:
Thanks for pointing it out, I guess not. There would be three options
1. Allow pushing to master for the bot
2. Change the workflow to create a PR for that change
3. Remove this step, let human bump versions

Reply:
The 2nd approach is taken